src/lib.rs

```diff
@@ -413,6 +418,144 @@ impl Contract {
             .collect()
     }
 
+    pub fn get_addon(&self, id: CommunityAddOnId) -> Option<CommunityAddOn> {
+        self.available_addons.get(&id)
+    }
+
+    pub fn get_addon_config(&self, id: CommunityAddOnConfigId) -> Option<CommunityAddOnConfig> {
+        self.addon_configurations.get(&id)
+    }
+
+    pub fn get_available_addons(&self) -> Vec<CommunityAddOn> {
+        self.available_addons.iter().map(|(_id, add_on)| add_on).collect()
+    }
+
+    // Only the contract admin and DevHub moderators
+    pub fn create_new_addon(&mut self, addon: CommunityAddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can create new add-ons");
+        }
+        if self.get_addon(addon.id.to_owned()).is_some() {
+            panic!("Add-on with this id already exists");
+        }
+        addon.validate();
+        self.available_addons.insert(&addon.id.clone(), &addon);
+    }
+
+    // ONLY FOR TESTING
+    pub fn delete_addon(&mut self, id: CommunityAddOnId) {
+        // Also delete from communities
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can delete add-ons");
+        }
+        let addon = self
+            .get_addon(id.clone())
+            .expect(&format!("Add-on with id `{}` does not exist", id))
+            .clone();
+
+        // let mut new_communities: UnorderedMap<CommunityHandle, Community> =
+        //     UnorderedMap::new(StorageKey::Communities);
+
+        // for (community_handle, mut community) in self.communities.iter() {
+        //     // Try to remove add on from community
+        //     community.remove_addon(addon.id.to_owned());
+        //     new_communities.insert(&community_handle, &community);
+        // }
+        // self.communities = new_communities;
+        self.available_addons.remove(&addon.id);
+    }
+
+    pub fn update_addon(&mut self, input: CommunityAddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can edit add-ons");
+        }
+        self.available_addons.insert(&input.id.clone(), &input);
+    }
+
+    pub fn get_community_addon_configs(
+        &self,
+        handle: CommunityHandle,
+    ) -> Vec<CommunityAddOnConfig> {
```

Comment:
This method won't be needed as the list of configs will be part of the Community struct.